### DISCUSSION TITLE
Should the CRD Validation Webhook be removed and its logic be moved?

### DISCUSSION THREAD

**Discussion by tylerslaton** [operator-framework/rukpak#388] [Polls]
There was a recent point brought up around the scalability of the [CRD Validation Webhook](https://github.com/operator-framework/rukpak/tree/main/cmd/crdvalidator). Say there are 10,000 CR's of a particular CRD on cluster. When we go to update that CRD, there will be something in the order of 10,000 requests querying those CRs and need to resolve within the timeout limit of around 60 seconds. 

The alternative to using the current webhook implementation is to instead do this logic inside of RukPak's BundleInstance controller to avoid the scalability issue and be more inline with eventual consistency. I figure we can leave this up to a poll to make a clear decision on this and move forward. When voting, please leave a comment to explain why.

**Comment by joelanford** [comment]
Is there an option for "yeah that sounds problematic, but not sure what to do about it yet"? :)

**Reply by tylerslaton** [reply]
I'm in a similar boat. I figure we can talk about it here in the comment section and then vote when we feel ready to.
